### PR TITLE
chore(core): scrutinizer issues

### PR DESCRIPTION
**Pull request by jeabakker**

### CODE REVIEW

File: docs/appendix/upgrade-notes/4.2-to-4.3.rst [hunk 1]
```diff
@@ -74,6 +77,10 @@ Class functions
 ~~~~~~~~~~~~~~~
 
 * ``\ElggEntity->deleteRelationships()`` use ``\ElggEntity->removeAllRelationships()``
+* ``\ELggUser->getFriends()`` use ``\ElggUser->getEntitiesFromRelationship()``
```
Comment (jdalsem) [inline docs/appendix/upgrade-notes/4.2-to-4.3.rst:80]: uppercase EL??

File: engine/classes/Elgg/Database/Plugins.php [hunk 2]
```diff
@@ -419,38 +419,33 @@ public function invalidateCache($plugin_id) {
...
+				'name' => 'title',
+				'value' => $plugin_id,
+			],
+			'limit' => 1,
+			'distinct' => false,
+		]);
 
-			return null;
-		};
+		if ($plugins) {
```
Comment (jdalsem) [inline engine/classes/Elgg/Database/Plugins.php]: maybe if (empty($plugins)) return null?

File: engine/lib/configuration.php [hunk 3]
```diff
@@ -86,12 +86,16 @@ function elgg_get_root_path() {
...
+ *
+ * @param string $path         The path
+ * @param bool   $append_slash Add trailing slash
  *
  * @return string
+ * @since 4.3
  */
-function elgg_get_engine_path() {
-	return Paths::elgg() . 'engine';
+function elgg_sanitize_path(string $path, bool $append_slash = true): string {
```
Comment (jdalsem) [inline engine/lib/configuration.php:97]: maybe elgg_sanitize_file_path?

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: docs/appendix/upgrade-notes/4.2-to-4.3.rst (hunk 1 above)
Comment (jeabakker) [reply docs/appendix/upgrade-notes/4.2-to-4.3.rst]: oops, fixes

File: engine/lib/configuration.php (hunk 3 above)
Comment (jeabakker) [reply engine/lib/configuration.php:97]: no

**Pull request merged**